auction-app-frontend/src/utils/helperFunctions.js

```diff
@@ -3,39 +3,40 @@ import {
   EMAIL_VALIDATOR,
   PASSWORD_LENGTH,
   NAME_VALIDATOR,
-  FORM_TYPES
+  FORM_TYPES,
+  COOKIE_PREFIX
 } from './constants';
+import Cookies from "js-cookie";
 
 export const getTotalPages = (products, size) => {
   return Math.ceil(products?.totalElements / size);
 };
 
 export const calculateTimeLeft = (product) => {
-  const startDate = new Date(product.startDate);
+  const currentDate = new Date();
   const endDate = new Date(product.endDate);
 
   const differenceInDays = Math.round(
-    (endDate.getTime() - startDate.getTime()) / (1000 * 60 * 60 * 24)
+    (endDate.getTime() - currentDate.getTime()) / (1000 * 60 * 60 * 24)
   );
 
   const differenceInWeeks = Math.floor(differenceInDays / 7);
 
-  if (differenceInWeeks > 0 && differenceInDays > 0) {
-    return `${differenceInWeeks} weeks ${differenceInDays} days`;
+  const remainingDays = differenceInDays % 7;
+
+  if (differenceInWeeks >= 0 && differenceInDays >= 0) {
+    return `${differenceInWeeks} weeks ${remainingDays} days`;
   }
   return;
 };
 
 export const getJwtFromCookie = () => {
-  const prefix = 'auction_app';
-  const allCookies = document.cookie.split(';');
-
-  for (let i = 0; i < allCookies.length; i++) {
-    let cookie = allCookies[i].trim();
+  const prefix = COOKIE_PREFIX;
+  const allCookies = Cookies.get();
```

Comment:
Cookies can be retrieved by name https://github.com/js-cookie/js-cookie#:~:text=Cookies.get(%27name%27)%20//%20%3D%3E%20%27value%27%0ACookies.get(%27nothing%27)%20//%20%3D%3E%20undefined